server/src/test/java/org/elasticsearch/index/query/TermsQueryBuilderTests.java

```diff
@@ -316,5 +323,16 @@ public void testTypeField() throws IOException {
         builder.doToQuery(createShardContext());
         assertWarnings(QueryShardContext.TYPES_DEPRECATION_MESSAGE);
     }
-}
 
+    @Override
+    protected QueryBuilder parseQuery(XContentParser parser) throws IOException {
+        QueryBuilder query = super.parseQuery(parser);
+        assertThat(query, CoreMatchers.instanceOf(TermsQueryBuilder.class));
+
+        TermsQueryBuilder termsQuery = (TermsQueryBuilder) query;
+        if (termsQuery.termsLookup() != null && termsQuery.termsLookup().type() != null) {
```

Comment:
maybe add an isTypeless() abstraction like you did for MLT?

Reply:
👍